config/runtime/server.go

```diff
@@ -422,31 +426,48 @@ func newInlineBackend(evalCtx *hcl.EvalContext, backends map[string]backendDefin
 		}
 	}
 	if inlineBlock == nil {
-		return nil, nil, errorMissingBackend
+		return nil, nil, "", errorMissingBackend
 	}
 
 	if err := validateInlineScheme(evalCtx, inlineBlock.Body, config.Backend{}); err != nil {
-		return nil, nil, err
+		return nil, nil, "", err
 	}
 
 	beConf := &config.Backend{}
 	diags = gohcl.DecodeBody(inlineBlock.Body, evalCtx, beConf)
 	if diags.HasErrors() {
-		return nil, nil, diags
+		return nil, nil, "", diags
 	}
 
+	var bodies []hcl.Body
+	var origin string
+
 	beConf, _ = defaultBackendConf.Merge(beConf)
 	if len(content.Blocks[0].Labels) > 0 {
 		beConf.Name = content.Blocks[0].Labels[0]
 		if beRef, ok := backends[beConf.Name]; ok {
-			beConf, _ = beRef.conf.Merge(beConf)
+			beConf, bodies = beRef.conf.Merge(beConf)
 		} else {
-			return nil, nil, fmt.Errorf("override backend %q is not defined", beConf.Name)
+			return nil, nil, "", fmt.Errorf("override backend %q is not defined", beConf.Name)
+		}
+	}
+
+	if bodies == nil {
+		bodies = append(bodies, beConf.Options)
+	}
+
+	bodies = append(bodies, inlineDef.Body())
+	bodies = append(bodies, inlineBlock.Body)
```

Comment:
This solution does not work. inlineBlock.Body equals to beRef.Options which is the first entry. I will do a rework here.

Reply:
And breaks other configurations.